core/src/unverified_blocks.rs

```diff
@@ -0,0 +1,97 @@
+use solana_ledger::entry::Entry;
+use solana_sdk::clock::Slot;
+use std::collections::{BTreeMap, BTreeSet, HashMap, VecDeque};
+
+struct UnverifiedBlockInfo {
+    parent: Slot,
+    // used as a correctness check
+    children: Vec<Slot>,
+    entries: Vec<Entry>,
+    fork_weight: u128,
+}
+
+#[derive(Default)]
+pub(crate) struct UnverifiedBlocks {
```

Comment:
@sakridge this is what I imagine will be shared/updated by your verification service, feel free to repurpose to your liking 😄